### PR TITLE
[v0.8 replacement 3/5] reconcile harness lifecycle

### PR DESCRIPTION
**Pull request by sethkarten**
## Replacement stack

Node 3/5 in a fresh append-only v0.8 replacement chain rooted at authoritative `main` commit `c2820d3679e6318f9c3d0155c823d6908b4e1faa`.

- **Base branch:** `v080/main-replacement-acp-meta`
- **Head branch:** `v080/main-replacement-harness-lifecycle`
- **Exact head:** `525a3374c54d1632bc23581a29d0e31c4c3b0c39`

## Scope

Exact coherent #2310 lifecycle/model-ID/skills delta: 7 paths; unrelated install newline drift excluded.

The old contributor PR branches remain untouched. This branch was published non-force from a fresh owned repository. Independent Luna semantic review **APPROVED** the full N1→N5 chain and found no structural scope leakage; authoritative-main renderer provenance is retained and `pyproject.toml` / `uv.lock` are unchanged across replacement nodes.

> [!WARNING]
> **TEST BLOCKED — do not merge or mark ready.** Authoritative `c2820d` has a stale `uv.lock` relative to `pyproject.toml`. A disposable `uv sync --locked` correctly refused before tests, without changing repository files. The only previously approved interpreter also cannot import current main because `pydantic_config` is absent. Testing/remediation continues on GitHub; this draft does not claim green.

No live, hosted, Docker, sandbox, model, or paid evaluation was run.

> [!NOTE]
> ### Reconcile harness lifecycle with skill validation and robust cleanup in `PrimeAgentHarness`
> - Introduces `Harness.resolved_skills()` in [harness.py](https://github.com/PrimeIntellect-ai/verifiers/pull/2326/files#diff-05e981bb3680a934fbc6865d9dff3b25fc692d0c9970684bf8d41f4aea7ee384) to validate skill folders and raise `ValueError` on duplicate basenames; `install_skills`, `PiHarness.launch`, and `PrimeAgentHarness._prepare` now delegate to this method.
> - `OpenClawHarness.launch` now writes `models.mode='replace'` in [openclaw/harness.py](https://github.com/PrimeIntellect-ai/verifiers/pull/2326/files#diff-d7278b9055ef0f18d323c2ac3010b475bc2122a03e99b44977384fab44c24362), preventing merges with built-in providers.
> - Rewrites `PrimeAgentHarness.cleanup` in [prime_agent/harness.py](https://github.com/PrimeIntellect-ai/verifiers/pull/2326/files#diff-20ba08f960d62f6e395d3eb05c3a6b051e309727747f31f1a381d7aa7396323b) to delete per-trace tmp then state root in two separate steps, perform a socket existence check, and raise `SandboxError` (with retry path context) on any socket, daemon stop, or deletion failure.
> - Adds test coverage for harness model ID isolation, skill staging collision detection, and the new cleanup behavior.
> - Risk: `cleanup` now raises `SandboxError` on previously-silent failures (e.g. unknown exit codes, timeouts), changing error propagation for callers.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 95cee3a.</sup>
> 
>

---

> [!NOTE]
> **Medium Risk**
> Prime Agent cleanup behavior and exception types change on rollout teardown; incorrect handling could leave daemon state or mis-attribute failures, though changes are heavily tested.
> 
> **Overview**
> Adds **`Harness.resolved_skills()`** so duplicate skill folder basenames fail before staging or CLI construction; **`install_skills`**, **Pi**, and **Prime Agent** now route skill paths through it.
> 
> **`PrimeAgentHarness.cleanup`** is reworked: socket check and daemon stop still gate deletion, but **per-trace tmp** and **state root** are removed in separate **`rm -rf`** steps with validated runtime results. Failures raise **`SandboxError`** (not **`RuntimeError`**) and include retry paths plus what was already removed. Stop failures block any **`rm`**.
> 
> New hermetic tests cover skill staging, OpenClaw/Pi model-ID intercept config, and Prime Agent cleanup edge cases (partial failure, idempotent retry, indeterminate runtime). OpenClaw/Pi comments clarify why the private **`intercept`** provider and OpenClaw **`models.mode: replace`** stay isolated from built-in catalogs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 95cee3a548d3e9b6481e40fce2b483bcd5e2c407. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: verifiers/v1/harnesses/openclaw/harness.py [hunk 1]
```diff
@@ -140,21 +141,10 @@ async def launch(
...
             },
             "models": {
-                "mode": "replace",
                 "providers": {
-                    "intercept": {
+                    provider: {
                         "baseUrl": endpoint,
                         "apiKey": "${OPENCLAW_INTERCEPT_KEY}",
-                        "api": "openai-responses",
-                        "authHeader": True,
```
[the remark this answers is not in the document]
Comment (sethkarten) [reply verifiers/v1/harnesses/openclaw/harness.py:149]: Fixed in ee15bfe66. Pi and OpenClaw now declare the exact opaque `ctx.model` under a private synthetic `intercept` provider, so bare, custom-prefix, and multi-slash IDs are always registered and selected without relying on built-in catalogs. Offline contracts cover five model-ID shapes; 10/10 passed.

File: verifiers/v1/harnesses/openclaw/harness.py [hunk 2]
```diff
@@ -140,21 +141,10 @@ async def launch(
...
-                        "models": [
```
[the remark this answers is not in the document]
Comment (sethkarten) [reply verifiers/v1/harnesses/openclaw/harness.py:150]: Fixed in ee15bfe66. OpenClaw now uses `models.mode: replace` with exactly one private `intercept` provider and explicit model declaration, preserving the Responses API/auth-header contract. This removes built-in provider endpoints and prevents model switches/fallbacks from bypassing interception. Offline config tests assert the isolated catalog.
